app/screens/settings/display_timezone_select/index.tsx

```diff
@@ -0,0 +1,149 @@
+
+import TimezoneRow from './timezone_row';
+
+const getStyleSheet = makeStyleSheetFromTheme((theme) => {
+    return {
+        container: {
+            flex: 1,
+            backgroundColor: theme.centerChannelBg,
+        },
+        searchBarInput: {
+            color: theme.centerChannelColor,
+            ...typography('Body', 100, 'Regular'),
+        },
+        searchBar: {
+            height: 38,
+            marginVertical: 5,
+        },
+    };
+});
+
+const EDGES: Edge[] = ['left', 'right'];
+const EMPTY_TIMEZONES: string[] = [];
+const ITEM_HEIGHT = 45;
+const VIEWABILITY_CONFIG = List.VISIBILITY_CONFIG_DEFAULTS;
+
+const keyExtractor = (item: string) => item;
+const getItemLayout = (_data: string[], index: number) => ({
+    length: ITEM_HEIGHT,
+    offset: ITEM_HEIGHT * index,
+    index,
+});
+
+type SelectTimezonesProps = {
+    selectedTimezone: string;
+    onBack: (tz: string) => void;
+}
+const SelectTimezones = ({selectedTimezone, onBack}: SelectTimezonesProps) => {
+    const intl = useIntl();
+    const serverUrl = useServerUrl();
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+
+    const [timezones, setTimezones] = useState<string[]>(EMPTY_TIMEZONES);
+    const [initialScrollIndex, setInitialScrollIndex] = useState<number>(0);
+    const [value, setValue] = useState('');
+
+    const filteredTimezones = (timezonePrefix: string) => {
+        if (timezonePrefix.length === 0) {
+            return timezones;
+        }
+
+        const lowerCasePrefix = timezonePrefix.toLowerCase();
+
+        return timezones.filter((t) => (
+            getTimezoneRegion(t).toLowerCase().indexOf(lowerCasePrefix) >= 0 ||
+            t.toLowerCase().indexOf(lowerCasePrefix) >= 0
+        ));
+    };
+
+    const onPressTimezone = useCallback((tzne: string) => {
+        onBack(tzne);
+        popTopScreen();
+    }, [onBack]);
+
+    const renderItem = ({item: timezone}: {item: string}) => {
+        return (
+            <TimezoneRow
+                onPressTimezone={onPressTimezone}
+                selectedTimezone={selectedTimezone}
+                timezone={timezone}
+            />
+        );
+    };
+
+    useEffect(() => {
+        // let's get all supported timezones
+        const getSupportedTimezones = async () => {
+            try {
+                const client = NetworkManager.getClient(serverUrl);
+                const allTzs = await client.getTimezones();
+                if (Array.isArray(allTzs)) {
+                    setTimezones(allTzs);
+                    const timezoneIndex = allTzs.findIndex((timezone) => timezone === selectedTimezone);
+                    if (timezoneIndex > 0) {
+                        setInitialScrollIndex(timezoneIndex);
+                    }
+                }
+            } catch (error) {
+                // eslint-disable-next-line no-console
+                console.log('FAILED TO GET ALL TIMEZONES', error);
+            }
+        };
```

Comment:
we should make this a remote action even if it does not store data in the db

Reply:
corrected